Assets/HoloToolkit/Utilities/Scripts/Solvers/SolverSurfaceMagnetism.cs

```diff
@@ -0,0 +1,592 @@
+
+	/// <summary>
+	///   SurfaceMagnetism casts rays to Surfaces in the world align the object to the surface.
+	/// </summary>
+	public class SolverSurfaceMagnetism : Solver
+	{
+		#region public enums
+		public enum RaycastDirectionEnum
+		{
+			CameraFacing,
+			ToObject,
+			ToLinkedPosition
+		}
+		public enum RaycastModeEnum
+		{
+			Simple,
+			Box,
+			Sphere
+		}
+
+		public enum OrientModeEnum
+		{
+			None,
+			Vertical,
+			Full,
+			Blended
+		}
+        #endregion
+
+        #region public members
+        [Tooltip("LayerMask to apply Surface Magnetism to")]
+        public LayerMask MagneticSurface = 0;
+
+        [Tooltip("Max distance to check for surfaces")]
+		public float MaxDistance = 3.0f;
+		[Tooltip("Closest distance to bring object")]
+		public float CloseDistance = 0.5f;
+
+		[Tooltip("Offset from surface along surface normal")]
+		public float SurfaceNormalOffset = 0.5f;
+		[Tooltip("Offset from surface along ray cast direction")]
+		public float SurfaceRayOffset = 0;
+
+		[Tooltip("Surface raycast mode.  Simple = single raycast, Complex = bbox corners")]
+		public RaycastModeEnum raycastMode = RaycastModeEnum.Simple;
+
+		[Tooltip("Number of rays per edge, should be odd. Total casts is n^2")]
+		public int BoxRaysPerEdge = 3;
+
+		[Tooltip("If true, use orthographic casting for box lines instead of perspective")]
+		public bool OrthoBoxCast = false;
+
+		[Tooltip("Align to ray cast direction if box cast hits many normals facing in varying directions")]
+		public float MaximumNormalVariance = 0.5f;
+
+		[Tooltip("Radius to use for sphere cast")]
+		public float SphereSize = 1.0f;
+
+		[Tooltip("When doing volume casts, use size override if non-zero instead of object's current scale")]
+		public float VolumeCastSizeOverride = 0;
+
+		[Tooltip("When doing volume casts, use linked AltScale instead of object's current scale")]
+		public bool UseLinkedAltScaleOverride = false;
+
+		// This is broken
+		[Tooltip("Instead of using mesh normal, extract normal from tex coord (SR is reported to put smoothed normals in there)")]
+		bool UseTexCoordNormals = false;
+
+		[Tooltip("Raycast direction.  Can cast from head in facing dir, or cast from head to object position")]
+		public RaycastDirectionEnum raycastDirection = RaycastDirectionEnum.ToLinkedPosition;
+
+		[Tooltip("Orientation mode.  None = no orienting, Vertical = Face head, but always oriented up/down, Full = Aligned to surface normal completely")]
+		public OrientModeEnum orientationMode = OrientModeEnum.Vertical;
+
+		[Tooltip("Orientation Blend Value 0.0 = All head 1.0 = All surface")]
+		public float OrientBlend = 0.65f;
+
+		[HideInInspector]
+		public bool OnSurface;
+		#endregion
+
+		#region private members
+        private BoxCollider m_BoxCollider;
+        #endregion
+
+        protected override void Start()
+        {
+            base.Start();
+
+			if (raycastMode == RaycastModeEnum.Box)
+			{
+                m_BoxCollider = GetComponent<BoxCollider>();
```

Comment:
formatting in this class is bad too

Reply:
Same thing as the previous comment, it looks fine in VS. I'm not sure why this is happening. I'll reformat/tabify.